client/tests/integration_tests.rs

```diff
@@ -0,0 +1,13 @@
+/// These tests are enabled with the tpm-simulator-tests feature and only run against a TPM
+/// simulator. The docker container in this crate builds a simulator binary, or you may provide
+/// your own simulator binary by setting the TPM_RS_SIMULATOR env var for `cargo test` on the
+/// command line.
+///
+/// To run locally in the docker container:
+///  `cd client && docker compose run simulator_tests`
+///
+/// These tests must be run with `--test-threads=1`, because they use a single TCP port.
+
+// Separate module to avoid us running these tests under normal `cargo test` runs.
+#[cfg(feature = "tpm-simulator-tests")]
+mod simulator_tests;
```

Comment:
can't we just move `client/tests/simulator_tests/mod.rs` to `client/tests/simulator_tests.rs` place a 
`#![cfg(feature = "tpm-simulator-tests")]` on line 1, delete everything else in this directory and it should still work exactly the same, with less nested structure.

Reply:
Amusing that actually works, we might want to keep the comments of this file in `client/tests/simulator_tests.rs`

Reply:
Ah yeah sorry, I didn't mention this in the PR description, but I did actually already try this and it doesn't work. :(

Reply:
Oh wait, it might work, it might have failed because I had it after the doc comment and that is a fatal error - I didn't notice that. Let me try again.

Reply:
Yup, it works! Updated the PR - I'll squash and merge. Thank you so much for the quick review!